### ISSUE TITLE
Report emails can get sent to the wrong email addresses, if multiple items are running in cron.

### ISSUE THREAD

**Issue by cklosowski** [easydigitaldownloads/edd-email-reports#10]
### Expected behavior
I expect the reports to only go to EDD admin emails.

### Actual behavior
Due to the cron that this runs on for daily reports, it's possible that there is an instance of the EDD_Emails class being instantiated at the same time as sending reports (for instance, sending renewal emails)

### Steps to reproduce the behavior
1) Setup a subscription that needs to have a renewal reminder sent.
2) Setup a report to go out daily as well.
3) 
etc, etc

### Information (if a specific version is affected):
PHP Version: All

Email Reports Version (or branch): `master`

EDD Version (or branch): All

WordPress Version: All

Any other relevant information:
This is actually a pretty edge case, as email reports uses it's own cron event to trigger, so it would require that both the `edd_email_reports_daily_email` event and another event sending EDD emails happens at the same time. I'll cook up a PR to see if we can help avoid this singleton issue.

**Comment by cklosowski** [comment]
Looking through this, i don't think it's the email addresses causing a problem, I think it's this action:
`add_action( 'edd_email_send_before', 'edd_email_reports_change_email_template' );`

We need to remove it once we've sent the report, so that a customer doesn't end up getting the report filter template...so the chain of events would look like this:
1) EDD Email Reports runs
2) Runs this action which triggers this filter: `add_filter( 'edd_email_template', 'edd_email_reports_set_email_template' );`, which runs this for the filter:
```
function edd_email_reports_set_email_template( $template_name ) {
	return 'report';
}
```
3) Sending a reminder or something else in the EDD_Emails platform, and since we never removed the filter, it's sending the report template.

**Comment by cklosowski** [comment]
TL;DR; we aren't sending to the wrong email address, I believe we're sending the wrong template.

**Comment by dgoldak** [comment]
@cklosowski it would make since we are sending the wrong template, not the wrong email address because the subject line was "Subscription payment received". If needed, we could confirm on the customers site, that those customers had renewals at the time the report was sent.

**Comment by mintplugins** [comment]
Replicated using these steps on `master`:

1. Add this code somewhere it will run:
```
	function my_custom_edd_reports_email_tester() {
		edd_email_reports_send_daily_email();
		EDD()->emails->send( edd_get_admin_notice_emails(), 'THIS SHOULD NOT BE AN EMAIL SALES REPORT', 'THIS SHOULD NOT BE AN EMAIL SALES REPORT' );
	}
	add_action( 'admin_init', 'my_custom_edd_reports_email_tester' );
```

2. Check in MailHog (or another email logger) to see which emails went out. You should see 2:
- Daily Sales Report for Recurring Payments
- THIS SHOULD NOT BE AN EMAIL SALES REPORT

<img width="1188" alt="Screen Shot 2019-10-01 at 5 22 59 PM" src="https://user-images.githubusercontent.com/1638128/66001738-22a1a600-e470-11e9-92e8-629a35a99a4e.png">

3. Open the email titled `THIS SHOULD NOT BE AN EMAIL SALES REPORT`. It should not show an email report, but instead just say `THIS SHOULD NOT BE AN EMAIL SALES REPORT` in the body. It does not:

<img width="1166" alt="Screen Shot 2019-10-01 at 5 19 42 PM" src="https://user-images.githubusercontent.com/1638128/66001774-36e5a300-e470-11e9-82ec-d1b7bd5391b6.png">

Confirmed fixed on `issue/10` using the same steps.

<img width="992" alt="Screen Shot 2019-10-01 at 5 24 38 PM" src="https://user-images.githubusercontent.com/1638128/66001857-63012400-e470-11e9-8522-7d4cd37a0221.png">

I also confirmed that the email sales report email was successfully sent:
<img width="1161" alt="Screen Shot 2019-10-01 at 5 25 36 PM" src="https://user-images.githubusercontent.com/1638128/66001919-80ce8900-e470-11e9-998d-f6bcd32f1975.png">
